Shared/sdk/CScriptArgReader.h

```diff
@@ -1052,6 +1071,31 @@ class CScriptArgReader
     }
 
 protected:
+    void InternalReadPair(std::vector<std::pair<SString, SString>>& outPairs, int iIndex)
+    {
+        lua_pushnil(m_luaVM);
+        while (lua_next(m_luaVM, iIndex) != 0)
+        {
+            int keyType = lua_type(m_luaVM, -2);
+            int valueType = lua_type(m_luaVM, -1);
+            if (keyType == LUA_TSTRING)
```

Comment:
Don't forget about the fact, that `LUA_TNUMBER` can be converted to a string as well. As this is a common function, I think the conversion to string should be accepted here.